core/src/main/scala/com/databricks/labs/remorph/parsers/SqlErrorStrategy.scala

```diff
@@ -0,0 +1,86 @@
+package com.databricks.labs.remorph.parsers
+
+import org.antlr.v4.runtime._
+import org.antlr.v4.runtime.misc.IntervalSet
+
+/**
+ * Custom error strategy for SQL parsing <p> While we do not do anything super special here, we wish to override a
+ * couple of the message generating methods and the token insert and delete messages, which do not create an exception
+ * and don't allow us to create an error message in context. Additionally, we can now implement i18n, should that ever
+ * become necessary.</p>
+ *
+ * <p>At the moment, we require valid SQL as input to the conversion process, but if we ever change that strategy, then
+ * we can implement custom recovery steps here based upon context, though there is no improvement on the sync()
+ * method.</p>
+ */
+abstract class SqlErrorStrategy extends DefaultErrorStrategy {
+
+  // Note that it is not possible to get this error from the current grammar, we would have to do an inordinate
+  // amount of mocking to raise this. It isn't worth the effort.
+  // $COVERAGE-OFF$
+  override protected def reportNoViableAlternative(recognizer: Parser, e: NoViableAltException): Unit = {
+    val tokens = recognizer.getInputStream
+    var input: String = null
+    if (tokens != null)
+      if (e.getStartToken.getType == Token.EOF) input = "<EOF>"
+      else input = tokens.getText(e.getStartToken, e.getOffendingToken)
+    else input = "<unknown input>"
+    val msg = new StringBuilder()
+    msg.append("could not process ")
+    msg.append(escapeWSAndQuote(input))
+    recognizer.notifyErrorListeners(e.getOffendingToken, msg.toString(), e)
+  }
+  // $COVERAGE-ON$
+
+  override protected def reportInputMismatch(recognizer: Parser, e: InputMismatchException): Unit = {
+    val msg = new StringBuilder()
+    msg.append(getTokenErrorDisplay(e.getOffendingToken))
+    msg.append(" was unexpected ")
+    msg.append(generateMessage(recognizer, e))
+    msg.append("\nexpecting one of: ")
+    msg.append(buildExpectedMessage(recognizer, e.getExpectedTokens))
```

Comment:
use interpolated stings: `val msg = s"${getTokenErrorDisplay(e.getOffendingToken)}  was unexpected ..."` - more readable

Reply:
Will. do - @vil1 also suggested the same